### ISSUE TITLE
gp_minimize leads to non reproducible results

### ISSUE THREAD

**Issue by miguelvr** [scikit-optimize/scikit-optimize#720]
Hi guys,

First of all, I really like this initiative, congratulations.

Regarding the issue: I'm trying to perform hyperparameter search for a model I implemented in pytorch, but the optimal parameters for my `train_and_evaluate` function returned by `gp_minimize` don't correspond to the same value when I re-run the same function again.

At start, I thought it could be a random seed issue, but I think I took all the measures to deal with that by setting the random, numpy and pytorch's seed to the same value.

Any idea what this issue could be?

Here's a snippet of my code:
https://gist.github.com/miguelvr/c85585441fcf468057ca65c52a2fbfe5

Thanks!
Miguel

**Comment by miguelvr** [comment]
Nevermind. This is either my code or pytorch's issue
